setup.py

```diff
@@ -1,16 +1,17 @@
-import os
-import re
+# -*- coding: utf-8 -*-
+import textwrap
 
 from setuptools import setup
 
-
-def get_version():
-    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), "virtualenv.py")) as file_handler:
-        version_file = file_handler.read()
-    version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", version_file, re.M)
-    if version_match:
-        return version_match.group(1)
-    raise RuntimeError("Unable to find version string.")
-
-
-setup(version=get_version(), py_modules=["virtualenv"], setup_requires=["setuptools >= 40.6.3"])
+setup(
+    use_scm_version={
+        "write_to": "src/virtualenv/version.py",
+        "write_to_template": textwrap.dedent(
+            """
+             # coding: utf-8
+             from __future__ import unicode_literals
+             __version__ = {version!r}
```

Comment:
Presumably minor too: do you actually mean for this to change to being unicode on Py2?

Reply:
yes 😄